### PR TITLE
Add test to follow existing behavior

### LINKED ISSUE #1: Add test to guarantee the existing behavior
We can add some functional test to check existing behavior.
See https://docs.gradle.org/current/userguide/testing_gradle_plugins.html#functional-tests

### PR DESCRIPTION
**Pull request by kojilin**
Motivation:

Before start making change, good to add unit test to check existing behavior

Modifications:

* Upgrade gradle to 7.6.1(Need to modify code to upgrade to 8)
* Add unit tests
* Remove current gradle plugin publish config, we can add later
* Add thrift binary 0.17.0 first

Result:

Close #1

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -24,36 +25,17 @@ archivesBaseName = 'thrift-gradle-plugin'
...
-
-	from(['LICENSE', 'NOTICE']) {
-		into 'META-INF'
-	}
+	implementation gradleApi()
+	implementation localGroovy()
+	testImplementation "org.assertj:assertj-core:3.24.2"
+	testImplementation "org.junit.jupiter:junit-jupiter:5.9.2"
+	testImplementation "org.mockito:mockito-junit-jupiter:5.1.1"
+	testImplementation "org.mockito.kotlin:mockito-kotlin:4.1.0"
```
Comment (haocheng) [inline build.gradle]: nit: should we use the latest version of these testing libs?

* junit-jupiter: 5.9.3
* mockito-junit-jupiter: 5.4.0
* mockito-kotlin: 5.0.0

File: src/test/java/com/linecorp/thrift/plugin/ThriftPluginTest.java [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+        Files.createDirectories(projectDir.resolve("src/main/thrift"));
+        Files.copy(Path.of("src/test/resources/test.thrift"),
+                   projectDir.resolve("src/main/thrift/test.thrift"));
+
+        thriftPath = Paths.get("lib/thrift-0.17.0");
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = { "7.6" })
```
Comment (haocheng) [inline src/test/java/com/linecorp/thrift/plugin/ThriftPluginTest.java:61]: We are using `@ParameterizedTest` to support running tests for multiple versions of Gradle in the future?

File: src/test/java/com/linecorp/thrift/plugin/ThriftPluginTest.java [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+public class ThriftPluginTest {
+    @TempDir
+    Path projectDir;
+
+    private Path buildFile;
+
+    private Path thriftPath;
+
+    @BeforeEach
+    public void setup() throws IOException {
```
Comment (haocheng) [inline src/test/java/com/linecorp/thrift/plugin/ThriftPluginTest.java]: nit: We can also throw `Exception` in the tests?

File: src/test/java/com/linecorp/thrift/plugin/ThriftPluginTest.java (hunk 2 above)
Comment (kojilin) [reply src/test/java/com/linecorp/thrift/plugin/ThriftPluginTest.java:61]: Yes

**Review by haocheng — APPROVED** [review]

**Review by ikhoon — APPROVED** [review]
Thanks!

**Pull request merged**